pep-0544.txt

```diff
@@ -439,98 +448,123 @@ but a static type checker will handle them specially. Subclassing a protocol
-where abstractness is simply defined by having at least one abstract method
-being unimplemented. Protocol classes must be marked *explicitly*.
+If ``Protocol`` is included in the base class list, all the other base classes
+must be protocols. A protocol can't extend a regular class, see `rejected`_
+ideas for rationale. Note that rules around explicit subclassing are different
+from regular ABCs, where abstractness is simply defined by having at least one
+abstract method being unimplemented. Protocol classes must be marked
+*explicitly*.
 
 
-Generic and recursive protocols
--------------------------------
+Generic protocols
+-----------------
 
 Generic protocols are important. For example, ``SupportsAbs``, ``Iterable``
 and ``Iterator`` are generic protocols. They are defined similar to normal
 non-protocol generic types::
 
-  T = TypeVar('T', covariant=True)
-
   class Iterable(Protocol[T]):
       @abstractmethod
       def __iter__(self) -> Iterator[T]:
           ...
 
-Note that ``Protocol[T, S, ...]`` is allowed as a shorthand for
-``Protocol, Generic[T, S, ...]``.
+``Protocol[T, S, ...]`` is allowed as a shorthand for
+``Protocol, Generic[T, S, ...]``. Declaring variance is not necessary for
+protocol classes, since it can be inferred from a protocol definition.
+Example::
+
+  class Box(Protocol[T]):
+      content: T
+
+  x: Box[float]
+  y: Box[int]
+  x = y  # This is OK due to inferred covariance of the protocol 'Box'.
+
+
+Recursive protocols
+-------------------
 
 Recursive protocols are also supported. Forward references to the protocol
 class names can be given as strings as specified by PEP 484. Recursive
-protocols will be useful for representing self-referential data structures
+protocols are useful for representing self-referential data structures
 like trees in an abstract fashion::
 
   class Traversable(Protocol):
       leaves: Iterable['Traversable']
 
+Note that for recursive protocols, a class is considered a subtype of
+the protocol in situations where such decision depends on itself.
+Continuing the previous example::
+
+  class SimpleTree:
+      leaves: List['SimpleTree']
+
+  root: Traversable = SimpleTree()  # OK
+
+  class Tree(Generic[T]):
+      def __init__(self, value: T,
+                         leaves: List['Tree[T]']) -> None:
+          self.value = value
+          self.leaves = leaves
+
+  def walk(graph: Traversable) -> None:
+      ...
+  tree: Tree[float] = Tree(0, [])
+  walk(tree)  # OK, 'Tree[float]' is a subtype of 'Traversable'
+
 
 Using Protocols
 ===============
 
 Subtyping relationships with other types
 ----------------------------------------
 
-Protocols cannot be instantiated, so there are no values with
-protocol types. For variables and parameters with protocol types, subtyping
-relationships are subject to the following rules:
+Protocols cannot be instantiated, so there are no values whose
+*exact* type is a protocol. For variables and parameters with protocol types,
```

Comment:
Maybe use 'runtime type' instead of 'exact type' as it's less open to interpretation.